### PR TITLE
At runtime replace $SHARED_DIR in paths (via FilePathHelper)

### PR DESCRIPTION
**Pull request by KoviRobi**
This is done before checking if the file exists, if you want a relative
path to a directory called "$SHARED_DIR", use "./$SHARED_DIR". The
reasoning for this is that this way, examples always appear as they are
intended to, even if there is a directory called "$SHARED_DIR" where
Mandelbulber was launched.

The following files have paths in them that I didn't know how to patch:
- Graeme McLaren  collection - license Creative Commons  (CC-BY 4.0)/mandelbulb_eye 001.fract
- Robert Pancoast collection - license Creative Commons (CC-BY 4.0)/box-deuce-deuce.fract
- Robert Pancoast collection - license Creative Commons (CC-BY 4.0)/menger-fatbackstrokin.fract
- mandelbox_menger_with_textures.fract
- monte carlo global illumination.fract

(From a grep -rl '[A-Z]:\\\|/home\|/usr' deploy/share/mandelbulber2/,
there might be more.)

Related to #719 

### This pullrequest changes
- Code change: mandelbulber2/src/files.cpp
- Change hardcoded paths in mandelbulber2/deploy/share/mandelbulber2

### CODE REVIEW

**Comment by buddhi1980** [comment]
By the way your code produces build errors caused by usage of SHARED_DIR (as I wrote it can be undefined for portable version)
```
g++ -c -pipe -ffast-math -fopenmp -I/usr/include/gsl -O3 -std=gnu++1y -D_REENTRANT -Wall -W -fPIC -DUSE_EXR -DUSE_TIFF -DUSE_SNDFILE -DUSE_OPENCL -DQT_MESSAGELOGCONTEXT -DQT_NO_DEBUG -DQT_UITOOLS_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_TESTLIB_LIB -DQT_CORE_LIB -DQT_TESTCASE_BUILDDIR='"/home/travis/build/buddhi1980/mandelbulber2/mandelbulber2/build"' -I../qmake -I. -I../../mandelbulber2 -I/opt/qt57/include -I/opt/qt57/include/QtUiTools -I/opt/qt57/include/QtWidgets -I/opt/qt57/include/QtMultimedia -I/opt/qt57/include/QtGui -I/opt/qt57/include/QtNetwork -I/opt/qt57/include/QtTest -I/opt/qt57/include/QtCore -I. -I../qt -isystem /usr/include/OpenEXR -isystem /usr/include/x86_64-linux-gnu -I/opt/qt57/mkspecs/linux-g++ -o files.o ../src/files.cpp
../src/files.cpp: In function ‘QString FilePathHelper(const QString&, const QStringList&)’:
../src/files.cpp:444:46: error: ‘SHARED_DIR’ was not declared in this scope
  newPath.replace(QRegExp("^\\$SHARED_DIR"), (SHARED_DIR));
```
When you correct code as I proposed I will pull this request.

File: mandelbulber2/src/files.cpp [hunk 1]
```diff
@@ -437,12 +437,16 @@ bool FileExists(const QString &path)
 
 QString FilePathHelper(const QString &path, const QStringList &pathList)
 {
-	// QString newPath;
+	// Prefer $SHARED_DIR to a file in a directory called "$SHARED_DIR", so that
+	// examples don't depend on the launch directory
+	// For a relative path starting with "$SHARED_DIR", use "./$SHARED_DIR"
+	QString newPath = path;
+	newPath.replace(QRegExp("^\\$SHARED_DIR"), (SHARED_DIR));
```
[the remark this answers is not in the document]
Comment (KoviRobi) [reply mandelbulber2/src/files.cpp]: Thanks, no need for the wrapping parenthesis due to it not being a macro!

**Comment by KoviRobi** [comment]
I added a comment to the GitHub pull-request template so that people adding examples know about this variable, not sure where else should I document it.

**Comment by buddhi1980** [comment]
Thanks @KoviRobi for your contribution

**Pull request merged**